### PR TITLE
Add mixed-reliability QUIC

### PR DESCRIPTION
**Pull request by oteffahi**
## Description
This PR adds support for mixed reliability QUIC links. It is supported on both QUIC and Unsecure QUIC (reliable UDP) links.

It is enabled by adding `mixed_rel=1` or `mixed_rel=auto` to the endpoint metadata.
ex: `quic/localhost:7447?mixed_rel=1`, `udp/localhost:7447?rel=1;mixed_rel=1`.
- Default is set to `false` to keep behavior consistent with previous versions.
- `auto` variant negotiates if `mixed_rel` is enabled or not depending on the peer's config/support.

### What does this PR do?
Send best effort traffic over datagrams and reliable traffic over QUIC streams, both over the same QUIC connection. This feature is compatible with multistream QUIC.

Depends on #2430

## Implementation details:
- Creates two distinct Zenoh links that use the same QUIC connection.
- Performs the Zenoh handshake over the reliable link. No Zenoh handshake is performed over the best-effort one.
- The two links are considered independent, however when one is closed the other is also closed to avoid scheduling further messages on it.
- Each link has its own Zenoh pipeline. The streamed QUIC link can also enable the multistream pipeline.
- Each link manages its own lease and sends/receives its own Zenoh `KeepAlive` messages.
- The two links only count as one with regards to multilink and the `transport/unicast/max_links` config.
- This feature is currently NOT compatible with lowlatency transport.

---
## 🏷️ Label-Based Checklist

Based on the labels applied to this PR, please complete these additional requirements:

**Labels:** `new feature`

## 🆕 New Feature Requirements

Since this PR adds a new feature:

- [x] **Feature scope documented** - Clear description of what the feature does and why it's needed
- [x] **Minimum necessary code** - Implementation is as simple as possible, doesn't overcomplicate the system
- [x] **New APIs well-designed** - Public APIs are intuitive, consistent with existing APIs
- [x] **Comprehensive tests** - All functionality is tested (happy path + edge cases + error cases)
- [x] **Examples provided** - Usage examples in code comments or separate example files
- [x] **Documentation added** - New docs explaining the feature, its use cases, and API
- [x] **Feature flag considered** - Can the feature be enabled/disabled for gradual rollout?
- [x] **Performance impact assessed** - Memory, CPU, storage implications measured
- [x] **Integration tested** - Feature works with existing features

**Consider:** Can this feature be split into smaller, incremental PRs?

**Instructions:**
1. Check off items as you complete them (change `- [ ]` to `- [x]`)
2. The PR checklist CI will verify these are completed

*This checklist updates automatically when labels change, but preserves your checked boxes.*

### CODE REVIEW

File: io/zenoh-link-commons/src/quic/utils.rs [hunk 1]
```diff
@@ -586,3 +647,94 @@ impl From<QuicAuthId> for LinkAuthId {
...
+        self
+    }
+}
+
+pub(crate) fn parse_mixed_reliability_config(metadata: Metadata<'_>) -> ZResult<bool> {
+    let Some(s) = metadata.get(Metadata::MIXED_RELIABILITY) else {
+        return Ok(false);
+    };
+
+    let Ok(mixed_rel) = s.parse::<u8>() else {
```
Comment (wyfo) [inline io/zenoh-link-commons/src/quic/utils.rs]: use a simple `match` to improve readability

File: io/zenoh-transport/src/unicast/universal/transport.rs [hunk 2]
```diff
@@ -157,47 +152,49 @@ impl TransportUnicastUniversal {
...
+                        &asl.link.link,
+                        asl.link.config.priorities.clone(),
+                        asl.link.config.reliability,
+                    ));
+                }
             })
             .await?;
         }
+
+        let res = {
```
Comment (wyfo) [inline io/zenoh-transport/src/unicast/universal/transport.rs]: use a join

File: io/zenoh-transport/src/unicast/universal/transport.rs [hunk 3]
```diff
@@ -271,45 +259,58 @@ impl TransportUnicastTrait for TransportUnicastUniversal {
...
         // Wrap the link
-        let (link, ack) = link.unpack();
+        let (link, ack, associated_link) = link.unpack();
         let (mut link, consumer) =
             TransportLinkUnicastUniversal::new(self, link, &self.priority_tx);
 
+        // Handle associated link (if mixed-reliability link)
+        let (associated_link, al_consumer) = associated_link
+            .map(|l| TransportLinkUnicastUniversal::new(self, l, &self.priority_tx))
+            .map_or((None, None), |(l, c)| (Some(l), Some(c)));
```
Comment (DenisBiryukov91) [inline io/zenoh-transport/src/unicast/universal/transport.rs]: why mapping a single option to a pair of options ?

File: io/zenoh-transport/src/unicast/universal/link.rs [hunk 4]
```diff
@@ -214,18 +226,70 @@ impl TransportLinkUnicastUniversal {
...
+                    cancellation_token,
+                    #[cfg(feature = "stats")]
+                    stats,
+                )
+                .await
+            })
+        }))
+        .await;
+        res.unwrap()?;
+    } else {
```
[the remark this answers is not in the document]
Comment (oteffahi) [reply io/zenoh-transport/src/unicast/universal/link.rs:261]: Out of the scope of this PR. Relevant for #2189 (cc @wyfo)

File: io/zenoh-transport/src/unicast/universal/link.rs [hunk 5]
```diff
@@ -284,52 +352,170 @@ async fn rx_task(
...
+            Poll::Pending
+        })
+        .await
+    }
+}
+
+impl Drop for TimeoutTracker {
+    fn drop(&mut self) {
+        self.0.task.get().unwrap().abort();
     }
```
[the remark this answers is not in the document]
Comment (oteffahi) [reply io/zenoh-transport/src/unicast/universal/link.rs:523]: Out of the scope of this PR. Relevant for #2189 (cc @wyfo)

File: io/zenoh-links/zenoh-link-udp/src/unicast.rs [hunk 6]
```diff
@@ -161,40 +163,56 @@ impl LinkUnicastTrait for LinkUnicastUdp {
         match &self.variant {
             LinkUnicastUdpVariant::Connected(link) => link.close().await,
             LinkUnicastUdpVariant::Unconnected(link) => {
-                link.close(self.src_addr, self.dst_addr).await
+                link.close(__self.src_addr, __self.dst_addr).await
```
[the remark this answers is not in the document]
Comment (oteffahi) [reply io/zenoh-links/zenoh-link-udp/src/unicast.rs]: It compiles because it's introduced by the `async_trait` proc-macro. I'll still remove it cuz it's confusing.

File: io/zenoh-transport/src/unicast/universal/transport.rs (hunk 3 above)
Comment (oteffahi) [reply io/zenoh-transport/src/unicast/universal/transport.rs]: I agree, it's unnecessary.
Fixed it.

File: io/zenoh-transport/src/unicast/universal/link.rs [hunk 7]
```diff
@@ -284,52 +352,170 @@ async fn rx_task(
...
+                        #[cfg(feature = "stats")]
+                        stats,
+                        &pool,
+                    ))
+                    .await
+            })
+        }))
+        .await;
+        res.unwrap().transpose()?;
+        Ok(())
```
[the remark this answers is not in the document]
Comment (oteffahi) [reply io/zenoh-transport/src/unicast/universal/link.rs:396]: Out of the scope of this PR. Relevant for https://github.com/eclipse-zenoh/zenoh/pull/2189 (cc @wyfo)

File: io/zenoh-links/zenoh-link-udp/src/reliability.rs [hunk 8]
```diff
@@ -29,33 +30,50 @@ use zenoh_result::ZResult;
...
+        ));
+        if is_mixed_rel {
+            let best_effort = Arc::new(LinkUnicastQuicDatagram::new(
+                quic_conn,
+                src_addr,
+                endpoint.clone().into(),
+                LinkAuthId::Udp,
+                UDP_LOCATOR_PREFIX,
+                None,
+            ));
```
[the remark this answers is not in the document]
Comment (oteffahi) [reply io/zenoh-links/zenoh-link-udp/src/reliability.rs:69]: `dst_locator` in the connect code-path should keep whatever was used to establish this link (`connect` config content or scouted endpoint), therefore I think it should be kept as is. This follows the same pattern as in the `zenoh-link-quic_datagram` crate.
